marlowe-dashboard-client/src/Contract/State.purs

```diff
@@ -130,16 +140,65 @@ mkInitialState walletDetails currentSlot followerAppId { chParams, chHistory } =
             # regenerateStepCards currentSlot
             # selectLastStep
 
-updateState :: Slot -> Array TransactionInput -> State -> State
-updateState currentSlot transactionInputs state =
+-- Note 1: We filter out PK parties from the participants of the contract. This is because
+-- we don't have a design for displaying them anywhere, and because we are currently only
+-- using one in a special case (in the Escrow with Collateral contract), where it doesn't
+-- make much sense to show it to the user anyway. If we ever want to use PK parties for
+-- other purposes, we will need to rethink this.
+-- Note 2: In general there is no way to map parties to wallet nicknames. It is possible
```

Comment:
yesterday I learn about this: it would be nice to see if there is something similar for Cardano, I'll ask in the slack
https://twitter.com/BrantlyMillegan/status/1386768539965693955

Reply:
aha, yes, that sounds promising...